host/sgx/sgxquote.c

```diff
@@ -131,12 +213,67 @@ oe_result_t oe_sgx_qe_get_target_info(
     size_t opt_params_size,
     uint8_t* target_info)
 {
-    oe_result_t result = OE_FAILURE;
+    oe_result_t result = OE_UNEXPECTED;
     quote3_error_t err = SGX_QL_ERROR_UNEXPECTED;
 
-    OE_UNUSED(format_id);
-    OE_UNUSED(opt_params);
-    OE_UNUSED(opt_params_size);
+    if (!format_id || !target_info)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    if (oe_sgx_initialize_quote_ex_library() == OE_OK)
```

Comment:
Can you add comment to why getting target info from quote_ex is favor over _sgx_qe_get_target_info()?

Reply:
The two functions could be talking to different QEs. While the quote_ex always talks to the QE in AESM, _sgx_qe_get_target_info() could be talking to the DCAP in-proc QE.

Reply:
@shnwc I don't think that speaks to @yentsanglee's suggestion, which isn't about why there are two methods of getting the target info in this function. I think he's pointing out that it's not obvious from reading the code why `_sgx_qe_get_target_info()` is only called if `oe_sgx_initialize_quote_ex_library()` fails entirely. i.e. it looks like an app expecting to talk to the DCAP in-proc QE would get short-circuited and never called if the quote_ex library is loadable.

It would be useful to explain as a code comment here (and elsewhere where this pattern is repeated, like in `oe_sgx_qe_get_quote_size()`) why this is either acceptable, or not the case (e.g. we expect that `oe_sgx_initialize_quote_ex_library()` is the function that checks the whether or not to use an internal provider and will fail so that this code then falls through to the `_sgx_qe_get_target_info()` call?).

Reply:
Got it. Will restructure the code and add relevant comments.